tracy-client/src/lib.rs

```diff
@@ -46,6 +50,79 @@ macro_rules! span {
     }};
 }
 
+#[cfg(all(feature="static_span", not(feature="enable")))]
+#[macro_export]
+macro_rules! static_span {
+    () => {{}};
+    ($name:expr) => {{}};
+}
+
+#[cfg(all(feature="static_span", feature="enable"))]
+#[macro_export]
+macro_rules! static_span {
+    () => {{
+        const LINENO: u32 = line!();
+        const FILE: &'static str = file!();
+        const MODULE_PATH: &'static str = module_path!();
+        const FILE_C: &'static str = $crate::const_format::concatc!(
+            MODULE_PATH,
+            ":",
+            FILE,
+            "\0"
+        );
+
+        // Source: https://stackoverflow.com/questions/38088067/equivalent-of-func-or-function-in-rust
+        fn f() {}
+        const fn type_name_of<T>(_: &T) -> &'static str {
+            std::any::type_name::<T>()
+        }
+        const F_FN_NAME: &'static str = type_name_of(&f);
+        const FN_NAME_C: &'static str = $crate::const_format::concatc!(
+            $crate::const_format::Sliced(F_FN_NAME, 0..F_FN_NAME.len() - 3),
+            "\0"
+        );
+        const SRC_LOC: $crate::sys::___tracy_source_location_data = $crate::sys::___tracy_source_location_data {
+            name: FN_NAME_C.as_ptr() as *const _,
```

Comment:
Not entirely sure a default such as this is the best. For instance, this would produce a pretty poor experience if a person used `static_span` twice in the same function.